build failure on RaspberryPi

At 47fbcc9.  My previous successful build was at d0430a4.
```
raspberrypi% make
    CC src/llvm-ptls.o
/home/sachs/src/julia-master/src/llvm-ptls.cpp: In function ‘llvm::Instruction* {anonymous}::emit_ptls_tp(llvm::LLVMContext&, llvm::Value*, llvm::Type*, llvm::Instruction*)’:
/home/sachs/src/julia-master/src/llvm-ptls.cpp:145:65: error: ‘asm_str’ was not declared in this scope
     auto tp = InlineAsm::get(FunctionType::get(T_pint8, false), asm_str, "=r", false);
                                                                 ^
Makefile:140: recipe for target 'llvm-ptls.o' failed
make[1]: *** [llvm-ptls.o] Error 1
Makefile:97: recipe for target 'julia-src-release' failed
make: *** [julia-src-release] Error 2
```

--- a/src/julia_threads.h
+++ b/src/julia_threads.h
@@ -81,7 +81,7 @@ typedef struct {
 
 // This includes all the thread local states we care about for a thread.
 #define JL_MAX_BT_SIZE 80000
-typedef struct _jl_tls_states_t {
+struct _jl_tls_states_t {
     struct _jl_gcframe_t *pgcstack;
     size_t world_age;
     struct _jl_value_t *exception_in_transit;
@@ -130,7 +130,7 @@ typedef struct _jl_tls_states_t {
     int finalizers_inhibited;
     arraylist_t finalizers;
     jl_gc_mark_cache_t gc_cache;
-} jl_tls_states_t;
+};
 
 // Update codegen version in `ccall.cpp` after changing either `pause` or `wake`
 #ifdef __MIC__

--- a/src/llvm-ptls.cpp
+++ b/src/llvm-ptls.cpp
@@ -136,6 +136,7 @@ static Instruction *emit_ptls_tp(LLVMContext &ctx, Value *offset, Type *T_ppjlva
 #elif defined(_CPU_X86_)
     const char *asm_str = "movl %gs:0, $0";
 #else
+    const char *asm_str = nullptr;
     assert(0 && "Cannot emit thread pointer for this architecture.");
 #endif
     if (!offset) {

--- a/src/tls.h
+++ b/src/tls.h
@@ -16,7 +16,9 @@ static inline unsigned long JL_CONST_FUNC jl_thread_self(void)
 #endif
 }
 
-typedef struct _jl_tls_states_t *jl_ptls_t;
+typedef struct _jl_tls_states_t jl_tls_states_t;
+
+typedef jl_tls_states_t *jl_ptls_t;
 
 #ifdef __cplusplus
 extern "C" {

